.github/workflows/test.yml

```diff
@@ -138,6 +138,90 @@ jobs:
           - name: npm install
             run: npm install
           - name: run tests
-            run: npm run test
+            run: npm run test:unit
       env:
           CI: true
+
+  frontend-e2e-tests:
+    runs-on: ubuntu-latest
+    name: Front-end E2E tests
+    strategy:
+      matrix:
+        php-version: [ '8.1' ]
+        nextcloud-version: [ 'master' ]
+        node-version: [ '20' ]
+    services:
+      mysql-service:
+        image: mariadb:10.5
+        env:
+          MYSQL_ROOT_PASSWORD: my-secret-pw
+          MYSQL_DATABASE: nextcloud
+          MYSQL_USER: nextcloud
+          MYSQL_PASSWORD: nextcloud
+        ports:
+          - 3306:3306
+        options: >-
+          --health-cmd="mysqladmin ping"
+          --health-interval=10s
+          --health-timeout=5s
+          --health-retries=3
+    steps:
+      - name: Set up php ${{ matrix.php-version }}
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: ${{ matrix.php-version }}
+          extensions: ctype,curl,dom,gd,iconv,intl,json,mbstring,openssl,posix,sqlite,xml,zip,gmp,oci8
+          coverage: xdebug
+      - name: Checkout Nextcloud
+        run: git clone https://github.com/nextcloud/server.git --recursive --depth 1 -b ${{ matrix.nextcloud-version }} nextcloud
+      - name: Check out the app
+        uses: actions/checkout@v3
+        with:
+          path: nextcloud/apps/twofactor_webauthn
+      - name: Install Nextcloud
+        run: php -f nextcloud/occ maintenance:install --database mysql --database-host 127.0.0.1 --database-name nextcloud --database-user nextcloud --database-pass nextcloud --admin-user admin --admin-pass admin
```

Comment:
https://github.com/ChristophWurst/setup-nextcloud :wink:

Reply:
Well :see_no_evil: I forgot about that one.

I'll refactor the action.

Reply:
Done